vktrace/vktrace_replay/vkreplay_vkreplay.cpp

```diff
@@ -1885,157 +1898,148 @@ void vkReplay::modifyMemoryTypeIndexInAllocateMemoryPacket(VkDevice remappedDevi
                    pPacketHeader1->packet_id == VKTRACE_TPI_VK_vkBindBufferMemory) {
             packet_vkBindImageMemory *bim = (packet_vkBindImageMemory *)(pPacketHeader1->pBody);
             if (traceAllocateMemoryRval == bim->memory) {
                 // A vkBIM/vkBBM binds memory allocated by this vkAM call.
-                foundBindMem = true;
-                bindMemIdx = i;
-                bindMemImage = bim->image;
                 if (pPacketHeader1->packet_id == VKTRACE_TPI_VK_vkBindImageMemory)
                     remappedImage = m_objMapper.remap_images(bim->image);
                 else
                     remappedImage = (VkImage)m_objMapper.remap_buffers((VkBuffer)bim->image);
-                bimMemoryOffset = bim->memoryOffset;
-            }
 
-            // Need to implement:
-            // Check for uses of the memory returned by this vkAM call in vkQueueBindSparse calls to find
-            // an image/buffer that uses the memory.
-            // ....
+                if (pPacketHeader1->packet_id == VKTRACE_TPI_VK_vkBindImageMemory && remappedImage &&
+                    replayImageToTiling[remappedImage] == VK_IMAGE_TILING_OPTIMAL) {
+                    // Skip optimal tiling image
+                    remappedImage = VK_NULL_HANDLE;
+                } else {
+                    foundBindMem = true;
+                    bindMemIdx = i;
+                    bindMemImage = bim->image;
+                    bimMemoryOffset = bim->memoryOffset;
+                }
+
+                // Need to implement:
+                // Check for uses of the memory returned by this vkAM call in vkQueueBindSparse calls to find
+                // an image/buffer that uses the memory.
+                // ....
+            }
         }
     }
 
     if (!foundBindMem) {
         // Didn't find vkBind{Image|Buffer}Memory call for this vkAllocateMemory.
-        // The memory may be mapped and written to by host, so find a mem index that
-        // has host writeable memory.
-        VkPhysicalDevice replayPhysicalDevice = replayPhysicalDevices[remappedDevice];
-        VkPhysicalDevice tracePhysicalDevice = tracePhysicalDevices[pPacket->device];
-
-        // First search through replay memoryRequirements array and look for mem idx with identical
-        // flags as used in trace vkAM.
-        uint32_t i;
-        for (i = 0; i < replayMemoryProperties[replayPhysicalDevice].memoryTypeCount; i++) {
-            if (traceMemoryProperties[tracePhysicalDevice].memoryTypes[pPacket->pAllocateInfo->memoryTypeIndex].propertyFlags ==
-                replayMemoryProperties[replayPhysicalDevice].memoryTypes[i].propertyFlags) {
-                *((uint32_t *)&pPacket->pAllocateInfo->memoryTypeIndex) = i;
-                break;
-            }
-        }
+        // This isn't an error - the memory is allocated but never used.
+        // So just use the index from the trace file and continue.
+        goto out;
```

Comment:
What if the index in the trace file is not valid on the replay device? The bad index will be used in the vkAllocateMemory and generate an error. Shouldn't an attempt be made to find a valid index?

Reply:
This is an error of the comments.  The memory allocated but never used will be skipped vkAM.
This helps to skip the vkAM calls for optimal images which will have their own vkAM call in replaying vkBindImageMemory().

I will submit a commit to correct the comments.